### PR TITLE
[APB-4282][GW] replace httpVerbs with HttpClient

### PR DESCRIPTION
**Pull request by geoffreywatson**

### CODE REVIEW

File: project/KenshooMonitoring.scala [hunk 1]
```diff
@@ -47,6 +47,7 @@ object KenshooMonitoringBuild extends Build {
       libraryDependencies ++= Seq(
           "org.scalatest" %% "scalatest" % "3.0.5" % "test",
           "org.mockito" % "mockito-core" % "2.23.4" % "test",
+        "org.scalamock" %% "scalamock" % "4.4.0" % Test,
```
Comment (sureshhmrc) [inline project/KenshooMonitoring.scala]: do we need this dependency now ?

File: README.md [hunk 2]
```diff
@@ -7,9 +7,7 @@ White-listing of URLs is possible for both.
 
 ## Note
 
-This library is currently intended for use by the Agent Services team.
-It has a direct dependency on http-verbs which makes the upgrade path to Play 2.6 or later more difficult.
-Until this is resolved we and PlatOps recommend that other teams avoid adding new dependencies on agent-kenshoo-monitoring.
+This library is currently intended for use by the Agent Services team. Since v4.3.0, to support Play 2.6 migrations the library no longer has a dependency on HttpVerbs. Now, you'll need an instance of HttpClient defined in scope.
```
Comment (sureshhmrc) [inline README.md]: and mention about implicit ExecutionContext too...
Also, no one cares about the README text i guess  :)

File: src/main/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClient.scala [hunk 3]
```diff
@@ -42,53 +44,60 @@ trait MonitoredWSHttp extends WSHttp with HttpAPIMonitor {
   val httpAPIs: Map[String, String]
   private lazy val apiNames = HttpAPINames(httpAPIs)
 
-  def monitorRequestsWithoutBodyIfUrlPatternIsKnown(method: String, url: String)(func: => Future[HttpResponse])(implicit hc: HeaderCarrier) = {
+  private def monitorRequestsWithoutBodyIfUrlPatternIsKnown(method: String, url: String)(func: => Future[HttpResponse])(implicit hc: HeaderCarrier) = {
```
Comment (sureshhmrc) [inline src/main/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClient.scala]: `monitorRequestsWithoutBodyIfUrlPatternIsKnown` , please use some simpler name like `monitor` ?

File: src/main/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClient.scala [hunk 4]
```diff
@@ -42,53 +44,60 @@ trait MonitoredWSHttp extends WSHttp with HttpAPIMonitor {
...
+      case Some(name) => {
+        monitor(name) {
+          func
+        }
+      }
     }
   }
 
-  def monitorRequestsWithBodyIfUrlPatternIsKnown[A](method: String, url: String)(func: => Future[HttpResponse])(implicit rds: Writes[A], hc: HeaderCarrier) = {
+  private def monitorRequestsWithBodyIfUrlPatternIsKnown[A](method: String, url: String)(func: => Future[HttpResponse])(implicit rds: Writes[A], hc: HeaderCarrier) = {
```
Comment (sureshhmrc) [inline src/main/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClient.scala]: same as above

File: src/test/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClientSpec.scala [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+import uk.gov.hmrc.play.test.UnitSpec
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ExecutionContext, Future}
+
+class MonitoredHttpClientSpec extends UnitSpec with MockitoSugar with BeforeAndAfterEach {
+  val http = mock[HttpClient]
+  val httpAPIs = Map("/test/endpoint" -> "testEndpoint")
+  val testEndpointTimer = mock[Timer]
+  val newTimer = mock[Timer]
```
Comment (sureshhmrc) [inline src/test/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClientSpec.scala]: two timers ?

File: src/test/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClientSpec.scala [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+    Mockito.reset(kenshooRegistry)
+    Mockito.reset(testEndpointTimer)
+  }
+
+
+  def greaterThanOrEqualTo(l: Long): ArgumentMatcher[lang.Long] = new ArgumentMatcher[lang.Long] {
+    override def matches(argument: lang.Long): Boolean = argument >= l
+  }
+
+  implicit val hc = HeaderCarrier()
```
Comment (sureshhmrc) [inline src/test/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClientSpec.scala]: you can move this to the top

File: src/test/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClientSpec.scala [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+  "doGet" should {
+
+    "time the request if the url is whitelisted" in new MonitoredHttpClientTest(http, kenshooRegistry, httpAPIs){
```
Comment (sureshhmrc) [inline src/test/scala/uk/gov/hmrc/agent/kenshoo/monitoring/MonitoredHttpClientSpec.scala]: just create `new MonitoredHttpClientTest(http, kenshooRegistry, httpAPIs)` instance once and re use it as anyway you are resetting the mock

File: README.md (hunk 2 above)
Comment (stevenmallen) [reply README.md]: > and mention about implicit ExecutionContext too...
> Also, no one cares about the README text i guess :)

I care about the README - please keep it up to date :)

**Pull request merged**